src/theory/sets/theory_sets_private.cpp

```diff
@@ -404,142 +276,49 @@ bool TheorySetsPrivate::assertFact( Node fact, Node exp ){
-          //must send as lemma
-          lemma.push_back( exp==d_true ? fact : NodeManager::currentNM()->mkNode( kind::IMPLIES, exp, fact ) );
-          return true;
-        }
-      }
-    }
-  }
-  return false;
-}
-void TheorySetsPrivate::assertInference( Node fact, Node exp, std::vector< Node >& lemmas, const char * c, int inferType ) {
-  d_keep.insert( exp );
-  d_keep.insert( fact );
-  if( assertFactRec( fact, exp, lemmas, inferType ) ){
-    Trace("sets-lemma") << "Sets::Lemma : " << fact << " from " << exp << " by " << c << std::endl;
-    Trace("sets-assertion") << "(assert (=> " << exp << " " << fact << ")) ; by " << c << std::endl;
-  }
-}
-
-void TheorySetsPrivate::assertInference( Node fact, std::vector< Node >& exp, std::vector< Node >& lemmas, const char * c, int inferType ){
-  Node exp_n = exp.empty() ? d_true : ( exp.size()==1 ? exp[0] : NodeManager::currentNM()->mkNode( kind::AND, exp ) );
-  assertInference( fact, exp_n, lemmas, c, inferType );
-}
-
-void TheorySetsPrivate::assertInference( std::vector< Node >& conc, Node exp, std::vector< Node >& lemmas, const char * c, int inferType ){
-  if( !conc.empty() ){
-    Node fact = conc.size()==1 ? conc[0] : NodeManager::currentNM()->mkNode( kind::AND, conc );
-    assertInference( fact, exp, lemmas, c, inferType );
+  else
+  {
+    return false;
   }
 }
-void TheorySetsPrivate::assertInference( std::vector< Node >& conc, std::vector< Node >& exp, std::vector< Node >& lemmas, const char * c, int inferType ) {
-  Node exp_n = exp.empty() ? d_true : ( exp.size()==1 ? exp[0] : NodeManager::currentNM()->mkNode( kind::AND, exp ) );
-  assertInference( conc, exp_n, lemmas, c, inferType );
-}
 
-void TheorySetsPrivate::split( Node n, int reqPol ) {
-  n = Rewriter::rewrite( n );
-  Node lem = NodeManager::currentNM()->mkNode( kind::OR, n, n.negate() );
-  std::vector< Node > lemmas;
-  lemmas.push_back( lem );
-  flushLemmas( lemmas );
-  Trace("sets-lemma") << "Sets::Lemma split : " << lem << std::endl;
-  if( reqPol!=0 ){
-    Trace("sets-lemma") << "Sets::Require phase "  << n << " " << (reqPol>0) << std::endl;
-    d_external.getOutputChannel().requirePhase( n, reqPol>0 );
-  }
+void TheorySetsPrivate::fullEffortReset()
+{
+  Assert(d_equalityEngine.consistent());
+  d_full_check_incomplete = false;
+  d_most_common_type.clear();
+  d_most_common_type_term.clear();
+  d_card_enabled = false;
+  d_rels_enabled = false;
+  // reset the state object
+  d_state.reset();
+  // reset the inference manager
+  d_im.reset();
+  // reset the cardinality solver
+  d_cardSolver->reset();
 }
 
 void TheorySetsPrivate::fullEffortCheck(){
   Trace("sets") << "----- Full effort check ------" << std::endl;
   do{
     Trace("sets") << "...iterate full effort check..." << std::endl;
-    Assert( d_equalityEngine.consistent() );
-    d_sentLemma = false;
-    d_addedFact = false;
-    d_full_check_incomplete = false;
-    d_set_eqc.clear();
-    d_set_eqc_list.clear();
```

Comment:
All other variables are reset in the new function fullEffortReset except `d_set_eqc_list` which I can not find in the new version. Is it removed or renamed ?

Reply:
This was an unused variable, I've removed it.